In [83]:
import pandas as pd

In [84]:
csvpath = "purchase_data copy.csv"
print(csvpath)
purchase_df = pd.read_csv(csvpath)
purchase_df.head()

purchase_data copy.csv


Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [85]:
purchase_df.count()

Purchase ID    780
SN             780
Age            780
Gender         780
Item ID        780
Item Name      780
Price          780
dtype: int64

In [86]:
##1. Player Count
# the number of total players
total_player = len(purchase_df["SN"].unique())
total_player_df = pd.DataFrame({"Total Players":[total_player]})
total_player_df

Total Players
0            576

In [87]:
##2. Purchasing Analysis (Total)
# the number of unique items
unique_items = purchase_unique_df = purchase_df["Item ID"].unique()
unique_number = len(unique_items)
# the number of purchase
purchase_number = len(purchase_df["Price"])
# the average price
average_price = purchase_df["Price"].mean()
average_price = "${:.2f}".format(average_price)
# total revenue
total_revenue = "${:.2f}".format(purchase_df["Price"].sum())   
# summary frame
summary = {"Number of Unique Items":[unique_number],"Average Price":[average_price],"Number of Purchases":[purchase_number],"Total Revenue":[total_revenue]}
summary_df = pd.DataFrame(summary)
summary_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780      $2379.77

In [88]:
##3. Gender Demographics
gender_data_df = purchase_df.groupby(["Gender"])
gender_player = gender_data_df["SN"].nunique()
gender_player_df = pd.DataFrame(gender_player)
gender_player_df["Percentage of Players"] = (gender_player_df["SN"]/total_player).map("{:.2%}".format)
gender_player_df = gender_player_df.rename(columns={"SN":"Total Count"})
gender_player_df

Total Count Percentage of Players
Gender                                                  
Female                          81                14.06%
Male                           484                84.03%
Other / Non-Disclosed           11                 1.91%

In [89]:
##4. Purchasing Analysis (Gender)
# Purchase Count
purchase_count = gender_data_df.count()["Purchase ID"]
# Average Purchase Price
average_pruchase_price_unf = gender_data_df["Price"].mean()
average_pruchase_price_f = average_pruchase_price_unf.map("${:.2f}".format)
# Total Purchase Value
gender_total_unf = gender_data_df["Price"].sum()
gender_total_f = gender_total_unf.map("${:.2f}".format)
# Avg Total Purchase per Person
gender_avg_person_unf = gender_total_unf/gender_player_df["Total Count"]
gender_avg_person_f = gender_avg_person_unf.map("${:.2f}".format)
# gender_summary = pd.DataFrame({"Purchase Count":[],"Average Purchase Price":[average_pruchase_price],"Total Purchase Value":[gender_total ],"Avg Total Purchase per Person":[]})
gender_summary = pd.DataFrame({"Purchase Count":purchase_count,"Average Purchase Price":average_pruchase_price_f,"Total Purchase Value":gender_total_f,"Avg Total Purchase per Person":gender_avg_person_f})
gender_summary

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [90]:
##5.Age Demographics
bins = [0,10,15,20,25,30,35,40,100]
group_name = ["<10","10-14","15-19","20-24","25-29","30-34","35-39","40+"]
purchase_df["Age Summary"] = pd.cut(purchase_df["Age"],bins,labels=group_name,right = False, include_lowest = True)
age_demo = purchase_df.groupby("Age Summary")["SN"].nunique()
age_demo_df = pd.DataFrame(age_demo)
age_demo_df = age_demo_df.rename(columns={"SN":"Total Count"})
age_demo_df["Percentage of Players"] = (age_demo_df["Total Count"]/total_player).map("{:.2%}".format)
age_demo_df

Total Count Percentage of Players
Age Summary                                   
<10                   17                 2.95%
10-14                 22                 3.82%
15-19                107                18.58%
20-24                258                44.79%
25-29                 77                13.37%
30-34                 52                 9.03%
35-39                 31                 5.38%
40+                   12                 2.08%

In [91]:
##6.Purchasing Analysis (Age)
# Purchase Count
age_purchase_count = purchase_df.groupby("Age Summary")["Price"].count()
# Average Purchase Price
age_average_price = purchase_df.groupby("Age Summary")["Price"].mean()
# Total Purchase Value
age_total = purchase_df.groupby("Age Summary")["Price"].sum()
# Avg Total Purchase per Person
age_avg_person = (age_total/age_demo_df["Total Count"]).map("${:.2f}".format)
age_summary = pd.DataFrame({"Purchase Count":age_purchase_count,"Average Purchase Price":age_average_price,"Total Purchase Value":age_total,"Avg Total Purchase per Person":age_avg_person})
age_summary["Average Purchase Price"] = age_summary["Average Purchase Price"].map("${:.2f}".format)
age_summary["Total Purchase Value"] = age_summary["Total Purchase Value"].map("${:.2f}".format)
age_summary

Purchase Count Average Purchase Price Total Purchase Value  \
Age Summary                                                               
<10                      23                  $3.35               $77.13   
10-14                    28                  $2.96               $82.78   
15-19                   136                  $3.04              $412.89   
20-24                   365                  $3.05             $1114.06   
25-29                   101                  $2.90              $293.00   
30-34                    73                  $2.93              $214.00   
35-39                    41                  $3.60              $147.67   
40+                      13                  $2.94               $38.24   

            Avg Total Purchase per Person  
Age Summary                                
<10                                 $4.54  
10-14                               $3.76  
15-19                               $3.86  
20-24                               $4.32  
25-29                               $3.81  
30-34                               $4.12  
35-39                               $4.76  
40+                                 $3.19

In [92]:
##7.Top Spenders
spender_df = purchase_df.groupby(["SN"])
# Purchase Count
spender_purchase_count = spender_df.count()["Age"]
# Average Purchase Price
spender_average = spender_df["Price"].mean()
# Total Purchase Value
spender_total = spender_df["Price"].sum()
# Spender Summary
summary_spender = pd.DataFrame({"Purchase Count":spender_purchase_count,"Average Purchase Price":spender_average,"Total Purchase Value":spender_total})
summary_spender_des = summary_spender.sort_values("Total Purchase Value",ascending = False)
summary_spender_des["Average Purchase Price"] = summary_spender_des["Average Purchase Price"].map("${:.2f}".format)
summary_spender_des["Total Purchase Value"] = summary_spender_des["Total Purchase Value"].map("${:.2f}".format)
summary_spender_des.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [93]:
##8.Most Popular Items
popular_df = purchase_df.groupby(["Item ID","Item Name"])
# Purchase Count
purchase_count = popular_df.count()["Price"]
# Items Price
items_price = total_value = popular_df["Price"].mean().map("${:.2f}".format)
# Total Purchase Value
total_value = popular_df["Price"].sum()
# Popularity Summary
summary_items = pd.DataFrame({"Purchase Count":purchase_count,"Items Price":items_price,"Total Purchase Value":total_value})
summary_populartity = summary_items.sort_values("Purchase Count",ascending = False)
summary_populartity["Total Purchase Value"] = summary_populartity["Total Purchase Value"].map("${:.2f}".format)
summary_populartity.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Items Price  \
Item ID Item Name                                                  
92      Final Critic                                       $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm       $4.23   
145     Fiery Glass Crusader                               $4.58   
132     Persuasion                                         $3.22   
108     Extraction, Quickblade Of Trembling Hands          $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [94]:
##9.Most Profitable Items
summary_profit = summary_items.sort_values("Total Purchase Value",ascending = False)
summary_profit["Total Purchase Value"] = summary_profit["Total Purchase Value"].map("${:.2f}".format)
summary_profit.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Items Price  \
Item ID Item Name                                                  
92      Final Critic                                       $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm       $4.23   
82      Nirvana                                            $4.90   
145     Fiery Glass Crusader                               $4.58   
103     Singed Scalpel                                     $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

*** Three Observable Trends ***
1. The Heroes of Pymoli is more popular among male players(652) than among female players(113), but female players spend more money for a single purchase($3.20) and for average total purchase($4.47)than male players($3.02,$4.07)
2. Players aged between 20-24(44.79%), 15-19(18.58%) and 25-29(13.37%) make up approximately three quarters of the total players amounts. Players aged between 35-39 spend more money for a single purchase($3.60) and for average total purchase($4.76) than other generations.
3. Player Lisosia93 spends the most money (total purchase value = $18.96). "Final Critic" and "Oathbreaker, Last Hope of the Breaking Storm" are the most popular and most profitable items (FC: purchase count = 13, total purchase value = $59.99; LHBS:purchase count = 12, total purchase value = $50.76)

